### PR TITLE
exclude symfony polyfills from php scoper during PHAR compilation

### LINKED ISSUE #550: Build fails with "Call to undefined function get_debug_type()"
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | 3.12.2@42e0a50
| PHP version | 7.4.16
| Platform with version | Ubuntu

When trying to build with the latest 3.12.2 .phar Release, I get the error:

 ```bash
./box compile --no-restart -vvv
 ```

 ```bash
PHP Fatal error:  Uncaught Error: Call to undefined function _HumbugBox48cf944fc33a\Symfony\Component\Console\get_debug_type() in phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php:523
 ```

I've tried earlier releases back to 3.8.5 but all with the same error.
This only happens on our jenkins server, locally (a windows machine) works fine.
Unfortunately I can't try the global composer installation on jenkins since we get dependency conflicts.

<details>
 <summary>box.json</summary>
 
 ```json

 {
    "compression": "GZ",
    "main": "bin/console",
    "output": "dist/deployer.phar",
    "check-requirements": false,
    "exclude-composer-files": false,
    "dump-autoload": false,
    "finder": [
        {
            "in": ".",
            "exclude": [
                "dist",
                "vendor",
                "spec",
                "bin/phpspec*",
                "bin/phpunit*",
                "phpspec.yml",
                "phpunit.xml.dist"
            ]
        },
        {

            "in": "vendor",
            "notName": "/[A-Za-z0-9_-].xml.dist/",
            "exclude": [
                "Test",
                "Tests",
                "test",
                "tests",
                "doc"
            ]
        }
    ],
    "git-version": "git-version"
 }
 ```
</details>

<details>
 <summary>Output</summary>
 
 ```bash
`Box Requirements Checker
========================

> Using PHP 7.4.16
> PHP is using the following php.ini file:
  /etc/php/7.4/cli/php.ini

> Checking Box requirements:
  ✔ The application requires the version "^7.2" or greater.
  ✔ The application requires the extension "zlib".
  ✔ The application requires the extension "phar".
  ✔ The package "beberlei/assert" requires the extension "json".
  ✔ The package "beberlei/assert" requires the extension "simplexml".
  ✔ The package "nikic/php-parser" requires the extension "tokenizer".
  
                                                                                
 [OK] Your system is ready to run the application.                              
                                                                                

[debug] Current memory limit: "-1"
[debug] Checking BOX_ALLOW_XDEBUG
[debug] The Xdebug extension is not loaded

    ____            
   / __ )____  _  __
  / __  / __ \| |/_/
 / /_/ / /_/ />  <  
/_____/\____/_/|_|  
                    

Box version 3.8.5@e708d75 2020-07-28 07:29:31 UTC

 // Loading the configuration file "/var/lib/jenkins/workspace/Deployer         
 // Client/Phar Build/box.json".                                                

🔨  Building the PHAR "/var/lib/jenkins/workspace/Deployer Client/Phar Build/dist/deployer.phar"


PHP Fatal error:  Uncaught Error: Call to undefined function _HumbugBox48cf944fc33a\Symfony\Component\Console\get_debug_type() in phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php:523
Stack trace:
#0 phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php(516): _HumbugBox48cf944fc33a\Symfony\Component\Console\Application->doActuallyRenderThrowable(Object(UnexpectedValueException), Object(_HumbugBox48cf944fc33a\Symfony\Component\Console\Output\StreamOutput))
#1 phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php(491): _HumbugBox48cf944fc33a\Symfony\Component\Console\Application->doRenderThrowable(Object(UnexpectedValueException), Object(_HumbugBox48cf944fc33a\Symfony\Component\Console\Output\StreamOutput))
#2 phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php(84): _HumbugBox48cf944fc33a\Symfony\Component\Console\Application- in phar:///var/lib/jenkins/workspace/Deployer Client/Phar Build/box/vendor/symfony/console/Application.php on line 523`
 ```
</details>

### PR DESCRIPTION
**Pull request by DAcodedBEAT**
This PR resolves this issue around PHAR compilation:
```
PHP Fatal error:  Uncaught Error: Call to undefined function _HumbugBox48cf944fc33a\Symfony\Component\Console\get_debug_type()
```

The solution used is using a similar approach to https://github.com/humbug/php-scoper/pull/481

Resolves #550

### CODE REVIEW

**Comment by DAcodedBEAT** [comment]
@box-project / @humbug It looks like I need maintainer approval to run the required Github workflow, so can someone please run this and merge this contribution at their earliest convenience?

This is my first time contributing to this project so I'm not completely sure what else is required of a PR in order for it to be accepted.

**Review by zbtnot — APPROVED** [review]

**Comment by zbtnot** [comment]
Looks good to me! Maybe someone from @box-project or @humbug could take a look? 😃

**Review by nmattis — APPROVED** [review]

**Review by jasonlodice — APPROVED** [review]

**Comment by DAcodedBEAT** [comment]
@theofidry I see you are listed as a maintainer of this project.  Can I get this reviewed and hopefully merged?  (also going forward, what is the protocol to get things merged for this project?)

**Comment by theofidry** [comment]
@DAcodedBEAT sorry I cannot check in details atm. Right now there is two blocking issues for respectively PHP-Scoper (a content overall which no one can do but I) and Box (#571) which need to be dealt with. Once done I believe there is some (minor) cleanup to do and both projects can be released once again with full 8.1 support

**Comment by theofidry** [comment]
Thank you @DAcodedBEAT

**Pull request merged**
